dialogue-target/src/main/java/com/palantir/dialogue/RequestAttachments.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue;
+
+import javax.annotation.Nullable;
+
+public interface RequestAttachments {
+
+    <V> V put(RequestAttachmentKey<V> key, V value);
+
+    <V> V getOrDefault(RequestAttachmentKey<V> key, @Nullable V defaultValue);
```

Comment:
Should both methods be annotated `@Nullable`?

Reply:
I don't think we want to support putting null, unless you want to reulse this api for remove?

Reply:
If there's no existing value, won't `put` return null?

Reply:
oh I thought you wanted to annotate the value; yeah I should annotate the return, you're right.

Reply:
Done